dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -131,6 +136,34 @@ LRESULT CSearchBar::OnSearchButtonClicked(WORD wNotifyCode, WORD wID, HWND hWndC
         return 0;
     }
 
+    // See if we have a szFileName, our first character is not an asterisk,
+    // and we have room in MAX_PATH to add an asterisk at the beginning of szFileName then add it
+    if (wcscmp(pSearchStart->szFileName, L"") != 0)
+    {
+        len = wcslen(pSearchStart->szFileName);
+        if(len > 0)
```

Comment:
`wcscmp(x, L"") != 0` and `wcslen(x) > 0` both check the same thing, so this check is superfluous.
Since you need the `wcslen` call anyway, you can reduce it down to just the `len` check.

Reply:
Changed in accordance with your recommendations. Thanks.